src/components/vm/vmActions.jsx

```diff
@@ -56,6 +59,124 @@ import store from "../../store.js";
 
 const _ = cockpit.gettext;
 
+const onStart = (vm, setOperationInProgress) => domainStart({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    setOperationInProgress(false);
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to start"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onInstall = (vm, onAddErrorNotification) => domainInstall({ vm, onAddErrorNotification }).catch(ex => {
+    onAddErrorNotification({
+        text: cockpit.format(_("VM $0 failed to get installed"), vm.name),
+        detail: ex.message.split(/Traceback(.+)/)[0],
+        resourceId: vm.id,
+    });
+});
+
+const onReboot = (vm) => domainReboot({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to reboot"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onForceReboot = (vm) => domainForceReboot({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to force reboot"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onShutdown = (vm, setOperationInProgress) => domainShutdown({ name: vm.name, id: vm.id, connectionName: vm.connectionName })
+        .then(() => !vm.persistent && cockpit.location.go(["vms"]))
+        .catch(ex => {
+            setOperationInProgress(false);
+            store.dispatch(
+                updateVm({
+                    connectionName: vm.connectionName,
+                    name: vm.name,
+                    error: {
+                        text: cockpit.format(_("VM $0 failed to shutdown"), vm.name),
+                        detail: ex.message,
```

Comment:
These 10 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1069-20230621-112432-f0b9982b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)